engine/consensus/compliance/engine.go

```diff
@@ -149,9 +173,108 @@ func (e *Engine) ProcessLocal(event interface{}) error {
 // Process processes the given event from the node with the given origin ID in
 // a blocking manner. It returns the potential processing error when done.
 func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
-	return e.unit.Do(func() error {
-		return e.process(originID, event)
-	})
+	e.pendingEventSink <- &Event{
```

Comment:
Why pushing an event to a FIFO queue? Is there a reason not calling `processPendingEvent` immediately?

Reply:
Of course, the main reason is to build a pipeline which will stream events in non blocking manner to consumer without blocking networking layer. Calling `processPendingEvent` has a high probability of blocking.